cli/src/action/mod.rs

```diff
@@ -128,3 +139,48 @@ fn msg_from_io_error(err: IoError) -> String {
         _ => "Unknown I/O error".into(),
     }
 }
+
+// determine if a full file path or just a file name has been given for the private key
+fn determine_key_file_path(key: &str) -> Result<PathBuf, CliError> {
+    if key.contains('/') {
+        Ok(PathBuf::from(key))
+    } else {
+        let mut path = dirs::home_dir().ok_or_else(|| {
+            CliError::ActionError(
+                "failed to load signing key: unable to determine home directory".into(),
+            )
+        })?;
+        path.push("splinter");
+        path.push("keys");
+        path.push(key);
+
+        if path.exists() {
+            Ok(path)
+        } else {
+            path.set_extension("priv");
+            if path.exists() {
+                Ok(path)
+            } else {
+                Err(CliError::ActionError(
+                    "failed to load signing key: could not be found".into(),
+                ))
+            }
+        }
+    }
+}
+
+// build a signed json web token using the private key
+fn create_token(key: &str) -> Result<String, CliError> {
```

Comment:
The argument name here, key, is confusing without reading the implementation of the function.

Reply:
Fixed